app/shop.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list[int]
+    products: dict[str: int | float]
+
+    @classmethod
+    def create_shops(cls, shops: dict) -> list:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]